### PR TITLE
feat: implement peak memory usage measure

### LINKED ISSUE #42: Peak Memory Measure
https://stackoverflow.com/questions/669438/how-to-get-memory-usage-at-runtime-using-c/14927379

### PR DESCRIPTION
**Pull request by Samarium150**
Close #42

### CODE REVIEW

File: include/solver/helper.h [hunk 1]
```diff
@@ -12,8 +12,7 @@
 #include <functional>
 #include <string>
 
-namespace solver {
-namespace helper {
+namespace solver::helper {
```
Comment (pwyq) [inline include/solver/helper.h:15]: I'm not sure if this is a good practice, but if you do, please also update the closing comments to L109 (`// namespace solver::helper`

Comment (pwyq) [reply include/solver/helper.h:15]: also in the corresponding helper.cpp, L80

Comment (Samarium150) [reply include/solver/helper.h:15]: The change is suggested by clang-tidy. I just updated the closing comments.

File: CMakeLists.txt
Comment (pwyq) [inline CMakeLists.txt:1]:
```
if(WIN32)
    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall")
ELSEIF(UNIX)
    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wreorder")
ENDIF()
```

**Review by pwyq — CHANGES_REQUESTED** [review]

**Review by pwyq — APPROVED** [review]

**Pull request merged**